src/descriptor/mod.rs

```diff
@@ -613,6 +647,17 @@ impl<Pk: MiniscriptKey + ToPublicKey> Descriptor<Pk> {
     }
 }
 
+impl Descriptor<DescriptorPublicKey> {
+    /// Derives all wildcard keys in the descriptor using the supplied `path`
+    pub fn derive(&self, path: &[bip32::ChildNumber]) -> Descriptor<DescriptorPublicKey> {
```

Comment:
I am not that familiar with HD wallets, but does it make sense to all pubkeys be derived using the same path? For example, `sh( and (pk/1/*, pk/2/*))`, why should the * be the same for all pubkeys?

Reply:
The `*` only stands for the last two derivation steps in my use case (internal/external, address number in wallet). So ideally you just specify your already-derived xpubs for that particular multisig wallet and these keys are derived in lockstep (which is what you want to do imo). You never want to generate an address from `sh(and(A/1/12, B/0/15))` imo.

Reply:
Agreed with Sebastian on my specific usecase as well, but i have no particular opinion regarding the API.

Reply:
I was thinking more of a multi-party setting, but I guess in that case the users can call `translate_fn` directly. Since that is not common, we can have this API as is or perhaps rename it `derive_all_keys` to make it more explicit?

Reply:
But it only derives keys with an asterisk at the end, so that wouldn't be all that accurate either.